### PR TITLE
Revert List component to old API

### PR DESCRIPTION
**Pull request by jimbo**
## This PR is a:

[ ] New feature
[ ] Enhancement/Optimization
[x] Refactor
[ ] Bugfix
[ ] Test for existing code
[ ] Documentation

## Summary

When this pull request is merged, it will...

Change the `List` component's API, such that it only expects `items` to be an iterable, instead of a stricter, `Map`-compatible, two-dimensional iterable. This change will make it more expedient to pass collections of data straight from the API to `List` without having to transform them first.

## Additional information

Since we can't safely mandate that all collections consist of objects with an `id` property, `List` must also accept a key-generation function, `getItemKey`. This function will default to `({ id }) => id`, which plucks the `id` prop from a plain object, covering the most common case. For flat collections of strings, such as tags, users can supply an identity function; for more unusual collections where the primary key is something other than `id`, they can supply a custom function.

**This PR is marked [WIP] until this API change is approved. Once approved, instances of `List` in this repo have to be updated.**

### CODE REVIEW

File: packages/peregrine/src/validators/iterable.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const isIterable = obj => typeof obj[Symbol.iterator] === 'function';
+
+function optionalIterable(props, propName, componentName) {
+    const prop = props[propName];
+
+    if (prop != null && !isIterable(prop)) {
+        return new Error(
+            `Invalid prop \`${propName}\` supplied to \`${componentName}\`. Validation failed.`
```
Comment (DrewML) [inline packages/peregrine/src/validators/iterable.js]: Can we make this more specific? Doesn't seem to call out that we're expecting an iterable

Comment (jimbo) [reply packages/peregrine/src/validators/iterable.js]: @DrewML Updated. 👍

**Comment by DrewML** [comment]
Thanks for doing this! Good that it now works seamlessly with our payloads, but can be useable by others with differently-shaped data.

**Comment by jimbo** [comment]
No problem, it's just a different approach to the same solution.

Fixes #41.

**Pull request merged**